airflow/migrations/versions/0140_2_9_0_update_trigger_kwargs_type.py

```diff
@@ -38,13 +42,43 @@
 airflow_version = "2.9.0"
 
 
+def get_session() -> sa.orm.Session:
+    conn = op.get_bind()
+    sessionmaker = sa.orm.sessionmaker()
+    return sessionmaker(bind=conn)
+
 def upgrade():
-    """Update trigger kwargs type to string"""
+    """Update trigger kwargs type to string and encrypt"""
     with op.batch_alter_table("trigger") as batch_op:
         batch_op.alter_column("kwargs", type_=sa.Text(), )
 
+    if not context.is_offline_mode():
+        session = get_session()
+        try:
+            for trigger in session.query(Trigger).options(lazyload(Trigger.task_instance)):
+                trigger.kwargs = trigger.kwargs
+            session.commit()
+        finally:
+            session.close()
```

Comment:
i'm fuzzy on this... but are all upgrades supposed to be in a single transaction?

Reply:
Yeah, this is something I was a little suspect of also, but haven't had a chance to verify it yet. Let me do that now.

That said, this is a pattern we have in existing migrations already 🤷.

Reply:
So turns out this doesn't actually commit the DDL OR DML, the "real" commit happens when alembic does it in a higher scope. Should be good here.

Reply:
weird. i guess it's a different session.

Reply:
Yeah it behaves like a "child session", which is kinda cool.